rocketpy/mathutils/function.py

```diff
@@ -1088,13 +1090,18 @@ def __call__(self, *args):
             evaluated at all points in the list and a list of floats will be
             returned. If the function is N-D, N arguments must be given, each
             one being an scalar or list.
+        filename : str | None, optional
+            The path the plot should be saved to. By default None, in which case
+            the plot will be shown instead of saved. Supported file endings are:
+            eps, jpg, jpeg, pdf, pgf, png, ps, raw, rgba, svg, svgz, tif, tiff
+            and webp (these are the formats supported by matplotlib).
 
         Returns
         -------
         ans : None, scalar, list
         """
-        if len(args) == 0:
-            return self.plot()
+        if len(args) == 0 or (len(args) == 1 and filename != None):
+            return self.plot(filename=filename)
```

Comment:
Great comments! I will check out some of the issues you have raised. You are absolutely correct that some of our treatment so far of the `call -> plot` parameters is a bit unclear.

Reply:
> (1) I believe that `is not None` is a better practice than `!= None`

Makes sense, I didn't know that convention before.

> (2) Why are we checking `len(args) == 1` and not passing `*args` to the plot function? What is the desired behavior for this?

I tried to copy the previous existing behavior. Before my changes, this code block was:

```py
if len(args) == 0:
    return self.plot()
else:
    return self.get_value(*args)
```

As you can see, the plot was only returned when no args were given. So to keep with the old behavior, I instead check if no args are given *or* exactly one arg, which is the filename. Now thinking about this a bit more I notice this is stupid since len(arg) is not affected by filename, so I don't need to check the arg count at all. I will modify this again 🙃

I'm not 100% sure if all this works, I was confused due to the function using an arbitrary amount of arguments, which honestly makes development more difficult. From my testing with a Python interpreter, it seems like the filename can only be set if callers of the function need to explicitly call `filename="something"` to set it, and all other arguments are assigned to *args. So in my opinion having filename in addition to *args should work.

> (3) Due to current implementations, `return self.plot(*args, filename=filename)` **will introduce errors!**

The args were not passed previously either, see the already mentioned code block. I think this is a different issue not related to this pull request.